examples/ethereum_implementation.md

```diff
@@ -0,0 +1,163 @@
+
+The message lifecycle for Ethereum is outlined below:
+
+## Message Preparation
+
+1. In the web interface, a user composes a message, including the following fields:
+
+    Field | Description | Example
+    | - | - | - |
+    To: | The recipient public address | `0xd5ab4ce3605cd590db609b6b5c8901fdb2ef7fe6`
+    From: | The sender public address | `0x92d8f10248c6a3953cc3692a894655ad05d61efb`
+    Reply-To: | The public address responses should be sent to | `0x92d8f10248c6a3953cc3692a894655ad05d61efb`
+    Subject: | The message subject | 
+    Body: | The message body |
+    Public Key: | The recipient public key^ | `0x69d908510e355beb1d5bf2df8129e5b6401e1969891e8016a0b2300739bbb00687055e5924a2fd8dd35f069dc14d8147aa11c1f7e2f271573487e1beeb2be9d0`
+
+    ^Note: to determine the recipient public key, an existing transaction needs to be sent on the blockchain by that Ethereum account. For example, to determine Bob’s public key, Bob needs to have sent a transaction.
+
+1. Next, a user ‘sends’ the message. The web interface sends the message as a POST request to the Mailchain app.
+
+1. The app adds the following default fields:
+
+    Field | Description | Example
+    | - | - | - |
+    Date: | The RFC1123 date format | `2019-04-12T18:21:00+01:00`
+    Message-id: | A unique message id composed of 64 chars (32 bytes) + `@mailchain` | `002c40fb138807253554afc5161740ca3dade11db7e74e799c9f6091b904277cb9b839393802dc38b8a815615543@mailchain`
+    Content-Type: | As per RFC6532 content type for the contents of the message | text/plain; charset="UTF-8" 
+    Content-Transfer-Encoding: | The message body is encoded according to this field | `quoted-printable`
+
+1. A hash of the message payload (headers + body etc.) is created (default: `SHA3-256`).
+
+    [returns `message_payload_hash`]
+
+1.  
+    1. The message is encrypted using the recipient public key (default: `AES-256-CBC`). This output is a byte array.
+
+    1. The encrypted data byte array is hashed (default murmur3). 
+
+    [returns `encrypted_message_hash`]
+
+1. The encrypted message is uploaded (PUT) to storage with the following attributes:
+
+    File name: `message-id`-`encrypted_message_hash`, e.g. `002c5d4ba47ce66f9e4b1f36f35e50c357aded81dfb9b98a89b8a80d5ca347b2a16f08dc5d37d255378ddcf3380d-220426516c9b`
+
+    [File contents: encrypted bytes]
+
+    The file storage location is returned. E.g. https://mcx.mx/002c5d4ba47ce66f9e4b1f36f35e50c357aded81dfb9b98a89b8a80d5ca347b2a16f08dc5d37d255378ddcf3380d-220426516c9b
+    
+    [returns `message_location`]
+
+1. The `message_location` is encrypted using the recipient public key (default AES-256-CBC). This is a byte array.
+
+    [returns `encrypted_message_location`]
+
+    1. A protocol buffer (or protobuff) (https://developers.google.com/protocol-buffers) is created containing the following fields:
```

Comment:
protobuff is with a single 'f' protobuf. There are multiple examples

Reply:
Changed everywhere.